falkordb_gemini_kg/models/openai.py

```diff
@@ -0,0 +1,109 @@
+from .model import GenerativeModel, GenerativeModelConfig, GenerationResponse, FinishReason, GenerativeModelChatSession
+from openai import OpenAI
+
+
+class OpenAiGenerativeModel(GenerativeModel):
+
+    client: OpenAI = None
+
+    def __init__(
+        self,
+        model_name: str,
+        generation_config: GenerativeModelConfig | None = None,
+        system_instruction: str | None = None,
+    ):
+        self.model_name = model_name
+        self.generation_config = generation_config
+        self.system_instruction = system_instruction
+
+    def _get_model(self) -> OpenAI:
+        if self.client is None:
+            self.client = OpenAI()
+
+        return self.client
+
+    def with_system_instruction(self, system_instruction: str) -> "GenerativeModel":
+        self.system_instruction = system_instruction
+        self.client = None
+        self._get_model()
+
+        return self
+
+    def start_chat(self, args: dict | None = None) -> GenerativeModelChatSession:
+        return OpenAiChatSession(self, args)
+
+    def ask(self, message: str) -> GenerationResponse:
```

Comment:
having both `start_chat` and `ask` functions is a bit confusing, we should strive for a simple and compact API

Reply:
We require both since the chat holds the history and the ask does not.